models/migrations/v128.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/core"
+	"xorm.io/xorm"
+)
+
+func addIssueWatchModes(x *xorm.Engine) error {
+	type IssueWatch struct {
+		ID          int64                 `xorm:"pk autoincr"`
+		UserID      int64                 `xorm:"UNIQUE(watch) NOT NULL"`
+		IssueID     int64                 `xorm:"UNIQUE(watch) NOT NULL"`
+		Mode        models.IssueWatchMode `xorm:"NOT NULL DEFAULT 1"`
+		CreatedUnix timeutil.TimeStamp    `xorm:"created NOT NULL"`
+		UpdatedUnix timeutil.TimeStamp    `xorm:"updated NOT NULL"`
+		//since it it is not used anymore and has NOT NULL constrain
+		//it is altered to allow NULL - we can drop it later ...
+		IsWatching bool
+	}
+
+	if x.Dialect().DBType() == core.SQLITE {
```

Comment:
If you do this inside a transaction, you can:
- `CREATE TABLE temp.issue_watch_old AS SELECT * FROM issue_watch`
- `DROP TABLE issue_watch` (this will get rid of the constraints)
- `Sync2()`
- `INSERT INTO issue_watch SELECT * FROM temp.issue_watch_old`
- `COMMIT`
I think it's safer and cleaner, and since it's inside a transaction a rollback will leave the original table intact.
Mind the `temp.` prefix on the temporary table; that will save you the need for a `DROP`.

Reply:
Sadly MySQL had no support for [in-transaction DDL](https://stackoverflow.com/questions/4692690/is-it-possible-to-roll-back-create-table-and-alter-table-statements-in-major-sql) until very recently, otherwise we could use this simple method on all of the engines.